### PR TITLE
Users allendm bugfix taefkey

### PR DESCRIPTION
**Pull request by allendm-msft**
from vs2015u3 onwards dont set the taef regkey for run in parallel

### CODE REVIEW

**Comment by nigurr** [comment]
Can we add UT for this function. Rest looks good.

File: Tasks/VsTest/Helpers.ps1 [hunk 1]
```diff
@@ -80,10 +80,16 @@ function IsVisualStudio2015Update1OrHigherInstalled {
...
 		# checking for dll introduced in vs2015 update1
 		# since path of the dll will change in dev15+ using vstestversion>14 as a blanket yes
-		if(Test-Path -Path "$env:VS140COMNTools\..\IDE\CommonExtensions\Microsoft\TestWindow\TE.TestModes.dll")
+		$teModesDll = [io.path]::Combine("$env:VS140COMNTools", "..", "IDE", "CommonExtensions", "Microsoft", "TestWindow", "TE.TestModes.dll");
+		if(Test-Path -Path $teModesDll)
 		{
-			# ensure the registry is set otherwise you need to launch VSIDE
-			SetRegistryKeyForParallel $vsTestVersion
+			$devenvExe = [io.path]::Combine("$env:VS140COMNTools", "..", "IDE", "devenv.exe");
+			$devenvVersion = [System.Diagnostics.FileVersionInfo]::GetVersionInfo($devenvExe);
```
Comment (nigurr) [inline Tasks/VsTest/Helpers.ps1]: can this throw exception? if my build agent is not running as admin?

Comment (allendm-msft) [inline Tasks/VsTest/Helpers.ps1]: no.. can only throw FNF which should never occur

**Comment by allendm-msft** [comment]
UTs for dev15+ added in prev PR this one is just to verify with U3+. This is verified manually

**Comment by nigurr** [comment]
Looks good.

**Pull request merged**